src/transforms/aggregate.rs

```diff
@@ -63,19 +109,80 @@ type MetricEntry = (metric::MetricData, EventMetadata);
+    multi_map: HashMap<metric::MetricSeries, Vec<MetricEntry>>,
+    mode: AggregationMode,
 }
 
 impl Aggregate {
     pub fn new(config: &AggregateConfig) -> crate::Result<Self> {
         Ok(Self {
             interval: Duration::from_millis(config.interval_ms),
             map: Default::default(),
+            prev_map: Default::default(),
+            multi_map: Default::default(),
+            mode: config.mode.clone(),
         })
     }
 
     fn record(&mut self, event: Event) {
         let (series, data, metadata) = event.into_metric().into_parts();
 
+        match self.mode {
+            AggregationMode::Auto => match data.kind {
+                metric::MetricKind::Incremental => self.record_sum(series, data, metadata),
+                metric::MetricKind::Absolute => {
+                    self.map.insert(series, (data, metadata));
+                }
+            },
+            AggregationMode::Sum => self.record_sum(series, data, metadata),
+            AggregationMode::Latest | AggregationMode::Diff => match data.kind {
+                metric::MetricKind::Incremental => (),
+                metric::MetricKind::Absolute => {
+                    self.map.insert(series, (data, metadata));
+                }
+            },
+            AggregationMode::Count => self.record_count(series, data, metadata),
+            AggregationMode::Max | AggregationMode::Min => {
+                self.record_comparison(series, data, metadata)
+            }
+            AggregationMode::Mean | AggregationMode::Stdev => match data.kind {
+                metric::MetricKind::Incremental => (),
+                metric::MetricKind::Absolute => {
+                    if matches!(data.value, MetricValue::Gauge { value: _ }) {
+                        match self.multi_map.entry(series) {
+                            Entry::Occupied(mut entry) => {
+                                let existing = entry.get_mut();
+                                existing.push((data, metadata));
+                            }
+                            Entry::Vacant(entry) => {
+                                entry.insert(vec![(data, metadata)]);
+                            }
+                        }
+                    }
+                }
+            },
+        }
+
+        emit!(AggregateEventRecorded);
+    }
+
+    fn record_count(&mut self, series: MetricSeries, data: MetricData, metadata: EventMetadata) {
+        let existing = self.map.entry(series).or_insert_with(|| {
+            let mut new_data = data.clone();
+            *new_data.value_mut() = MetricValue::Counter { value: 0f64 };
+            (new_data, metadata.clone())
+        });
+        let mut count_data = data.clone();
+        *count_data.value_mut() = MetricValue::Counter { value: 1f64 };
+        if existing.0.kind == data.kind && existing.0.update(&count_data) {
+            existing.1.merge(metadata);
+        } else {
+            emit!(AggregateUpdateFailed);
+        }
+    }
```

Comment:
```suggestion
    fn record_count(
        &mut self,
        series: MetricSeries,
        mut data: MetricData,
        metadata: EventMetadata,
    ) {
        let mut count_data = data.clone();
        let existing = self.map.entry(series).or_insert_with(|| {
            *data.value_mut() = MetricValue::Counter { value: 0f64 };
            (data.clone(), metadata.clone())
        });
        *count_data.value_mut() = MetricValue::Counter { value: 1f64 };
        if existing.0.kind == data.kind && existing.0.update(&count_data) {
            existing.1.merge(metadata);
        } else {
            emit!(AggregateUpdateFailed);
        }
    }
```

I think the suggested code is identical in functionality but might avoid cloning. If it makes sense to you, we can commit it.